custom_components/waste_collection_schedule/waste_collection_schedule/source/kiedysmieci_info.py

```diff
@@ -0,0 +1,154 @@
+import datetime
+import ssl, urllib, json, requests
+from waste_collection_schedule import Collection
+from waste_collection_schedule.exceptions import (
+    SourceArgumentNotFoundWithSuggestions,
+)
+
+TITLE = "Kiedy śmieci"
+DESCRIPTION = "Source script for Kiedy śmieci, Poland"
+URL = "https://kiedysmieci.info"
+COUNTRY = "pl"
+TEST_CASES = {
+    "Nadolany, podkarpackie, sanocki, Bukowsko": {
+        "voivodeship": "podkarpackie",
+        "district": "sanocki",
+        "municipality": "Bukowsko",
+        "street": "Nadolany",
+    },
+    "Kędzierz, podkarpackie, dębicki, Dębica": {
+        "voivodeship": "podkarpackie",
+        "district": "dębicki",
+        "municipality": "Dębica",
+        "street": "Kędzierz",
+    },
+}
+
+API_URL = "https://cloud.fxsystems.com.pl:8078/odbiory_smieci/%s"
+
+ICON_MAP = {
+    "zmieszane": "mdi:trash-can",
+    "metale i tworzywa sztuczne": "mdi:recycle",
+    "papier i tektura": "mdi:file-outline",
+    "szkło": "mdi:glass-fragile",
+    "biodegradowalne": "mdi:leaf",
+}
+
+def get_json(url):
+    #workaround to establish ssl connection to this host
+    ssl_ctx = ssl.create_default_context()
+    ssl_ctx.set_ciphers("DEFAULT")
+
+    response = urllib.request.urlopen(API_URL % url, context=ssl_ctx)
+    return json.loads(response.read().decode("utf-8"))
+ 
+def get_municipalities():
+    try:
+        municipalities = get_json("dostepne_gminy/v5")["gminy"]
+    except:
+        municipalities = None
+
+    return municipalities
+
+municipalities = get_municipalities()
```

Comment:
I don't like this, as this does a web request every time someone calls the update_docu_links script.

Reply:
Ok, I changed it.